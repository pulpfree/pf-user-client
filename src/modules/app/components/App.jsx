import React, { Component } from 'react'

import { Redirect, Route } from 'react-router-dom'

import './app.sass'
import Header from './Header'
import Sites from '../../site/components/Sites'
import Users from '../../user/components/Users'

class App extends Component {


  render() {

    let isAuthed = true

    // console.log('location:', this.props.location)
    // console.log('split:', this.props.location.pathname.split('/'))
    // const pathPrts = this.props.location.pathname.split('/')

    if (!isAuthed) {
      // console.log('isAuthed:', isAuthed)
    // } else {
      // console.log('isAuthed:', isAuthed)
      return <Redirect to={'/login'} />
    }

    return (
      <div className="app">
        <Header props={this.props} />
        {/*<p className="app-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>*/}
        <Route
          component={Sites}
          path="/admin/sites"
        />
        <Route
          component={Users}
          path="/admin/users"
        />
      </div>
    )
  }
}

export default App
