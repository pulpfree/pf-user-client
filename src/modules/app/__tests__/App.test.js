/* eslint no-undef: "off" */
import React from 'react'
// import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import App from '../components/App'

// import { BrowserRouter as Router } from 'react-router-dom'

describe('components', () => {
  describe('<App />', () => {

    const appC = shallow(<App />)

    /*it('renders without crashing', () => {
      const div = document.createElement('div')
      ReactDOM.render(<Router><App /></Router>, div)
      ReactDOM.unmountComponentAtNode(div)
    })*/

    it('always renders a div', () => {
      const divs = appC.find('div')
      expect(divs.length).toBeGreaterThan(0)
    })

    it('has correct className', () => {
      expect(appC.find('div').hasClass('app')).toBe(true)
    })

    it('matches snapshot', () => {
      expect(appC).toMatchSnapshot()
    })

    it('has required Routes', () => {
      const routes = appC.find('Route')
      expect(routes.length).toBe(2)
    })

  })
})