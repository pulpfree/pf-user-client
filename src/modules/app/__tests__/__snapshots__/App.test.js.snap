// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <App /> matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          props={Object {}}
/>,
        <p
          className="app-intro"
>
          To get started, edit 
          <code>
                    src/App.js
          </code>
           and save to reload.
</p>,
        <Route
          component={[Function]}
          path="/admin/sites"
/>,
        <Route
          component={[Function]}
          path="/admin/users"
/>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "props": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "To get started, edit ",
            <code>
              src/App.js
</code>,
            " and save to reload.",
          ],
          "className": "app-intro",
        },
        "ref": null,
        "rendered": Array [
          "To get started, edit ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "src/App.js",
            },
            "ref": null,
            "rendered": "src/App.js",
            "type": "code",
          },
          " and save to reload.",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/admin/sites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/admin/users",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            props={Object {}}
/>,
          <p
            className="app-intro"
>
            To get started, edit 
            <code>
                        src/App.js
            </code>
             and save to reload.
</p>,
          <Route
            component={[Function]}
            path="/admin/sites"
/>,
          <Route
            component={[Function]}
            path="/admin/users"
/>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "props": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "To get started, edit ",
              <code>
                src/App.js
</code>,
              " and save to reload.",
            ],
            "className": "app-intro",
          },
          "ref": null,
          "rendered": Array [
            "To get started, edit ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "src/App.js",
              },
              "ref": null,
              "rendered": "src/App.js",
              "type": "code",
            },
            " and save to reload.",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/admin/sites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/admin/users",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
